core/src/main/scala/org/apache/spark/deploy/history/HistoryPage.scala

```diff
@@ -30,8 +31,17 @@ private[history] class HistoryPage(parent: HistoryServer) extends WebUIPage("")
     val requestedIncomplete = Option(request.getParameter("showIncomplete"))
       .getOrElse("false").toBoolean
 
-    val displayApplications = parent.getApplicationList()
-      .exists(isApplicationCompleted(_) != requestedIncomplete)
+    val displayApplications = {
+      val applicationList = parent.getApplicationList()
```

Comment:
Cannot close `KVStoreIterator` in `getApplicationList` because it return  an `iterator`